File: agent/CarRentalPortal.Core/Entities/Rental.cs
using System;

namespace CarRentalPortal.Core.Entities
{
    public class Rental
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int CarId { get; set; }
        public Car Car { get; set; }
        public int RentalBundleId { get; set; }
        public RentalBundle RentalBundle { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public DateTime RequestedOn { get; set; }
        public string Remarks { get; set; }
    }
}

File: agent/RentalApp/src/app/rentals/rental.ts
export interface RentalBundleVm {
	rentalBundles: RentalBundle[];
}

export interface RentalBundle {
	id: number;
	numberOfItems: number;
	status: RentalStatus;
	createdOn: Date;
	rentals: Rental[];
}

export interface RentalVm {
	rentals: Rental[];
}

export interface Rental {
	id: number;
	rentalBundleId: number;
	carId: number;
	carBrand: string;
	carModel: string;
	carYear: number;
	customerFullName: string;
	customerContactInfo: string;
	pickupDate: Date;
	returnDate: Date;
	status: RentalStatus;
}

export interface BundleRequest {
	rentals: RentalRequest[];
}

export interface RentalRequest {
	ownerId: number;
	customerId: number;
	carId: number;
	startDate?: Date;
	endDate?: Date;
	remarks?: string;
	isBundle: boolean;
}

export interface RentalResponse {
	bundleId: number;
	status: RentalStatus;
}

export enum RentalStatus {
	PENDING = 'Pending',
	ACCEPTED = 'Accepted',
	REJECTED = 'Rejected'
}